### ISSUE TITLE
Add HSTS headers to response of callable functions

### ISSUE THREAD

**Issue by mbleigh** [firebase/firebase-functions#862]
Since we want `.onCall` functions to be HTTPS-only, we should restrict them via HSTS headers.

**Comment by joaoaguiam** [comment]
Hi there, is there any news on this feature request?

**Comment by shauryaaher** [comment]
Yes please, this is important. Please make this feature available for everyone.

PS ~ Please make Cloud Functions available on the spark plan. It's really necessary as I'm not an enterprise developer.

**Comment by ishowta** [comment]
I'm not security friendly, so apologies if I'm wrong, but since functions are not web pages and are never opened directly by the user, isn't there a need to enable HSTS?

**Comment by cabljac** [comment]
I may be wrong, but I'm not sure what purpose HSTS headers would serve for onCall functions.

HSTS instructs browsers to always use HTTPS for future requests to a domain, but callable functions are invoked by Firebase SDKs, not navigated to by browsers.

Additionally, Cloud Functions and Cloud Run endpoints are already HTTPS-only at the infrastructure level no? There's no HTTP listener to upgrade from. Happy to be corrected if I'm missing something, and will reopen if so.

**Comment by ajperel** [comment]
I consulted with some more knowledgeable Firebase folks just to double check. HSTS does apply for XHR, but in our case it's just a latency helper. If you're using the Firebase SDKs we'll always use HTTPS anyway and HSTS _should_ never apply. Additionally as was called out in the last response our servers don't serve HTTP traffic just HTTPS so in the worst case we're saving people trying to use HTTP an error or redundant roundtrip. TL;DR adding HSTS headers wouldn't hurt, could help latency in some cases, but isn't needed from a security standpoint.
